### PR TITLE
Refactor tests for improved reporting, part 2

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

**Comment by kevinearls** [comment]
@objectiser @pavolloffay @jpkrohling please review

File: test/e2e/cassandra_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	jaegerPod, err := GetPod(namespace, "with-cassandra", "jaegertracing/all-in-one", fw.KubeClient)
+	require.NoError(t, err, "Failed to find pod starting with with-cassandra with image jaegertracing/all-in-one")
+	portForw, closeChan, err := CreatePortForward(namespace, jaegerPod.Name, []string{"16686", "14268"}, fw.KubeConfig)
+	require.NoError(t, err, "Failed to create PortForward")
+
+	defer portForw.Close()
+	defer close(closeChan)
+	err = SmokeTest("http://localhost:16686/api/traces", "http://localhost:14268/api/traces", "foobar", retryInterval, timeout)
+	require.NoError(t, err, "SmokeTest Failed")
+	}
```
Comment (objectiser) [inline test/e2e/cassandra_test.go]: nit: format issue

File: test/e2e/cassandra_test.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+func (suite *CassandraTestSuite) TestCassandraSparkDependencies()  {
+	storage := v1.JaegerStorageSpec{
+		Type: "cassandra",
+		Options: v1.NewOptions(map[string]interface{}{"cassandra.servers": cassandraServiceName, "cassandra.keyspace": "jaeger_v1_datacenter1"}),
+		CassandraCreateSchema:v1.JaegerCassandraCreateSchemaSpec{Datacenter:"datacenter1", Mode: "prod"},
+	}
+	err := sparkTest(t, framework.Global, ctx, storage)
+	require.NoError(t, err, "SparkTest failed")
+
```
Comment (objectiser) [inline test/e2e/cassandra_test.go]: nit: remove additional line

File: test/e2e/es_index_cleaner_test.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// +build elasticsearch
+
```
Comment (objectiser) [inline test/e2e/es_index_cleaner_test.go]: Wondering whether the es-index-cleaner test should just be included in the elasticsearch_suite?

File: test/e2e/production_test.go [hunk 4]
```diff
@@ -1,41 +1,50 @@
+// +build elasticsearch
+
```
Comment (objectiser) [inline test/e2e/production_test.go]: I think this smoke test for ES should be moved into elasticsearch test suite - so it is equivalent to the cassandra tests.

File: test/e2e/self_provisioned_elasticsearch_test.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	ctx.Cleanup()
+}
+
+func TestSelfProvisionedSuite(t *testing.T) {
+	suite.Run(t, new(SelfProvisionedTestSuite))
+}
+
+func (suite *SelfProvisionedTestSuite) TestSelfProvisionedESSmokeTest() {
+	// create jaeger custom resource
+	exampleJaeger := &v1.Jaeger{
```
Comment (objectiser) [inline test/e2e/self_provisioned_elasticsearch_test.go]: To be consistent with structure, shouldn't this be moved into a separate function?

File: test/e2e/self_provisioned_elasticsearch_test.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+				Type: "elasticsearch",
+			},
+		},
+	}
+	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, "simple-prod-collector", 1, retryInterval, timeout)
+	if err != nil {
+		fmt.Printf(">>>>>>>>>>>>>>>> Error %v\n", err)
```
Comment (objectiser) [inline test/e2e/self_provisioned_elasticsearch_test.go]: This seems to be new - is there any particular reason for the sleep?

File: test/e2e/elasticsearch_test.go [hunk 7]
```diff
@@ -51,3 +60,156 @@ func (suite *ElasticSearchTestSuite) TestSparkDependenciesES() {
...
+	require.NoError(t, err, "Error waiting for Cron Job")
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		flag, err := hasIndexWithPrefix("jaeger-")
+		return !flag, err
+	})
+	require.NoError(t, err, "TODO")
+}
+
+func getJaegerSimpleProdWitServerUrls() *v1.Jaeger {
```
Comment (objectiser) [inline test/e2e/elasticsearch_test.go]: nit: WithServerUrls

**Pull request merged**